dbms/src/Common/MyTime.cpp

```diff
@@ -1097,26 +1244,29 @@ bool MyTimeBase::isValid(bool allow_zero_in_date, bool allow_invalid_date) const
         }
     }
 
-    if (year >= 9999 || month > 12)
+    if (year > 9999 || month > 12)
```

Comment:
why do we change behavior here?

Reply:
Because cast(99991111.1111111 as datetime) is valid now,  and result is 9999-11-11.